packages/remix-express/server.ts

```diff
@@ -38,14 +50,59 @@ export type RequestHandler = (
 export function createRequestHandler({
   build,
   getLoadContext,
+  serverMiddleware,
   mode = process.env.NODE_ENV,
 }: {
   build: ServerBuild;
   getLoadContext?: GetLoadContextFunction;
+  serverMiddleware?: ServerMiddlewareFunction;
   mode?: string;
 }): RequestHandler {
   let handleRequest = createRemixRequestHandler(build, mode);
 
+  if (build.future.unstable_middleware) {
+    return async (
+      req: express.Request,
+      res: express.Response,
+      next: express.NextFunction
+    ) => {
+      let response: NodeResponse | undefined;
+      try {
+        if (serverMiddleware) {
+          let context = createMiddlewareStore();
+          let callRemix = async () => {
+            let request = createRemixRequest(req, res);
+            response = (await handleRequest(
+              request,
+              undefined,
+              context
+            )) as NodeResponse;
+            return response;
+          };
+          context.next = callRemix;
+
+          await serverMiddleware({ request: req, response: res, context });
+
+          if (!response) {
+            // User never called next(), so doesn't need to do any post-processing
+            response = await callRemix();
+          }
+          if (!res.headersSent) {
+            await sendRemixResponse(res, response);
+          }
+        } else {
+          let request = createRemixRequest(req, res);
+          response = (await handleRequest(request)) as NodeResponse;
+          await sendRemixResponse(res, response);
```

Comment:
If you turn on `unstable_middleware` and don't provide a `serverMiddleware` we just call through directly without any concept of `getLoadContext`